instrumentation/manager.go

```diff
@@ -266,10 +273,15 @@ func (m *manager[ProcessGroup, ConfigGroup]) handleProcessExecEvent(ctx context.
 		return err
 	}
 
-	err = inst.Load(ctx)
-	// call the reporter regardless of the load result - as we want to report the load status
-	reporterErr := m.handler.Reporter.OnLoad(ctx, e.PID, err, pg)
-	if err != nil {
+	loadErr := inst.Load(ctx)
+
+	// we need to track the instrumentation even if the load failed.
+	// consider a reporter which writes a persistent record for a failed/successful load
+	// we need to notify the reporter once that PID exists to clean up the resources
+	m.startTrackInstrumentation(e.PID, inst, pg, configGroup, loadErr == nil)
```

Comment:
if we had an error with the `Load`, it means that at this point the `inst` is not longer relevant. We will not need to call `Close` or apply config to it since it's not loaded. Thus, I recommend to consider using `nil` instead of `inst` in this case so that it doesn't get called accidentally and cause hard to find bug